### ISSUE TITLE
Attributify not working despite not using it

### ISSUE THREAD

**Issue by rehhouari** [windicss/windicss#320]
I'm using the Vite plugin, everything updated to the latest version. I turned on Attributify mode:
```ts
export default defineConfig({
  attributify: true,
...
```

and without even using it, it gives me this error:

```bash
3:53:13 PM [vite] Internal server error: _this._variants[i] is not a function
      at /home/rehhouari/Projects/web/templates/rapide/node_modules/.pnpm/windicss@3.1.1/node_modules/windicss/lib/index.js:11282:62
      at Array.map (<anonymous>)
      at /home/rehhouari/Projects/web/templates/rapide/node_modules/.pnpm/windicss@3.1.1/node_modules/windicss/lib/index.js:11282:18
      at Array.map (<anonymous>)
      at Processor.wrapWithVariants (/home/rehhouari/Projects/web/templates/rapide/node_modules/.pnpm/windicss@3.1.1/node_modules/windicss/lib/index.js:11277:23)
      at _gStyle (/home/rehhouari/Projects/web/templates/rapide/node_modules/.pnpm/windicss@3.1.1/node_modules/windicss/lib/index.js:11897:37)
      at /home/rehhouari/Projects/web/templates/rapide/node_modules/.pnpm/windicss@3.1.1/node_modules/windicss/lib/index.js:11924:53
      at Array.forEach (<anonymous>)
      at _loop_2 (/home/rehhouari/Projects/web/templates/rapide/node_modules/.pnpm/windicss@3.1.1/node_modules/windicss/lib/index.js:11924:23)
      at Processor.attributify (/home/rehhouari/Projects/web/templates/rapide/node_modules/.pnpm/windicss@3.1.1/node_modules/windicss/lib/index.js:11933:13)
```

If i use it, it gives me the same error.

I've searched and found only https://github.com/windicss/windicss/issues/308#issue-906791858

I assume this is because I'm using Alpine.js which uses attributes? (like x-data, x-text, etc)

**Comment by xorik** [comment]
Same error in 3.1.0 and 3.1.1 😞

**Comment by ahku** [comment]
Yeah, I had to downgrade to an older version in order to fix this issue.

**Comment by antfu** [comment]
Can you provide more info? Preferably a minimal reproduction, thanks.

**Comment by rehhouari** [comment]
> Can you provide more info? Preferably a minimal reproduction, thanks.

Working on it

**Comment by rehhouari** [comment]
@antfu I wasn't able to get a minimal reproduction, weird. I've a full-project that have the problem which you can clone from: https://github.com/rehhouari/rapide

`pnpm i` then `pnpm dev` as usual.

BTW, Does Windicss read the HTML Meta tags' attributes as attributify attributes?

See: those are meta tags not in the body

```bash
rehhouari@lubuntu:~/Projects/web/templates/rapide$ node_modules/windicss/cli/index.js -a index.html 
Matched files: [ 'index.html' ]
Output file: /home/rehhouari/Projects/web/templates/rapide/windi.css
rehhouari@lubuntu:~/Projects/web/templates/rapide$ cat windi.css 
...
[content~="no"] {
  content: "no";
}
[content~="#ffffff"] {
  content: "#ffffff";
}
[content~="/images/icons/icon-144x144.png"] {
  content: "/images/icons/icon-144x144.png";
}
[content~="{app_name}"] {
  content: "{app_name}";
}
[content~="default"] {
  content: "";
}
[content~="yes"] {
  content: "yes";
}
```

**Comment by rehhouari** [comment]
I've added `console.log(variants)` to the line that crash `node_modules/.pnpm/windicss@3.1.1/node_modules/windicss/lib/index.js:11283:62`
to see which attribute is causing the problem and I guess I was right? it's Alpine.js ones that start with`x-something`:

```bash
{ variants: [ 'hover' ] }
{ variants: [ 'disabled' ] }
{ variants: [ 'disabled' ] }
{ variants: [ 'disabled' ] }
{ variants: [ 'hover' ] }
{ variants: [ 'hover' ] }
{ variants: [ 'x' ] } <- this
4:06:33 PM [vite] Internal server error: _this._variants[i] is not a function <- i === 'x' 
```

**Comment by rehhouari** [comment]
I've made a pull request with the work around I've mentioned if you think it is fit
Used a for loop instead of only checking the first variant in the array

**Comment by rehhouari** [comment]
@xorik, @ahku a workaround is using a prefix:

for example:

```ts
// windicss.config
attributify: {
prefix: 'w:'
}
```
```html
<div w:bg="gray-50" ....
```

**Comment by xorik** [comment]
@rehhouari minimal demo is here:
https://github.com/xorik/vite-dark-bg-opacity, not sure if my issue is related or I should create a new issue.

**Comment by alexanderniebuhr** [comment]
one common thing @xorik & @rehhouari have is that it seems to break for both of them if the attribute has a dash `-` in it..

**Comment by xorik** [comment]
@alexanderniebuhr looks like an argument should ends on `-from` or `-to`, `-bottom`, `-top` (maybe something else) to cause the issue.
e.g.
```
<div bla-bla-to="red-200"/>
```

**Comment by rehhouari** [comment]
> one common thing @xorik & @rehhouari have is that it seems to break for both of them if the attribute has a dash `-` in it..

Could be cause the `x` in variants that is crashing WindiCSS is most likely part of Alpine.js' `x-text` attribute or such

**Comment by voorjaar** [comment]
should been fixed in v3.1.2
